pkg/database/database.go

```diff
@@ -240,6 +240,13 @@ func (db *Database) WriteUrls(
 	jsonPath string,
 	logs ...models.LogTuple,
 ) error {
+	if k8sObj == nil {
+		return errors.New("Cannot write log urls to the database when k8sObj is nil")
+	}
+	if len(logs) == 0 {
+		return errors.New("Cannot write log urls to the database when no logs are provided")
+	}
```

Comment:
The check before and this one are very domain specific, not that related with the database itself, because they are preconditions to continue with the database operations. What do you think about moving them to the parent caller, `Sink.WriteUrls`? We already have a precondition there in which the creation of URLs should be successful anyway, so we keep the business preconditions together.

Reply:
I think that it makes sense to move this check to the sink side. I'll add a comment to `Database.WriteUrls` to mention that if logs is empty or `nil`, any existing logs will be removed and will not be replaced. I do think that the we should keep the `k8sObj == nil` check though